### ISSUE TITLE
Cannot perform multipart object copy when object key contains spaces

### ISSUE THREAD

**Issue by edjames** [aws/aws-sdk-ruby#1241]
Hi

When I perform a copy/move operation on an object when the object key contains a space, and I specify a multipart copy, the operation fails. When I don't specify a multipart copy, the operation succeeds. The file I'm testing with is 6.6MB (big enough for a multipart copy).

I'm doing this to rename the file for a 3rd-party vendor.

This does work in v1 of the sdk, but not in v2. I'm using Ruby 2.3.0.

##### Version 1 (Good on version 1.66.0)

```
> source_key = 'users/79357/ej-HD 1.mp4'
=> "users/79357/ej-HD 1.mp4"
> target_key = 'users/79357/ej-hd-1.mp4'
=> "users/79357/ej-hd-1.mp4"
> source = AWS::S3.new.buckets['my-bucket'].objects[source_key]
=> <AWS::S3::S3Object:my-bucket/users/79357/ej-HD 1.mp4>
> source.move_to target
[AWS S3 200 1.269625 0 retries] copy_object(:bucket_name=>"my-bucket",:copy_source=>"/my-bucket/users/79357/ej-HD 1.mp4",:key=>"users/79357/ej-hd-1.mp4",:metadata_directive=>"COPY",:storage_class=>"STANDARD")
[AWS S3 204 0.137253 0 retries] delete_object(:bucket_name=>"my-bucket",:key=>"users/79357/ej-HD 1.mp4")
=> <AWS::S3::S3Object:my-bucket/users/79357/ej-hd-1.mp4>
> target.move_to source, use_multipart_copy: true, content_length: target.content_length
[AWS S3 200 1.05901 0 retries] head_object(:bucket_name=>"my-bucket",:key=>"users/79357/ej-hd-1.mp4")
[AWS S3 200 0.131331 0 retries] initiate_multipart_upload(:bucket_name=>"my-bucket",:copy_source=>"/my-bucket/users/79357/ej-hd-1.mp4",:key=>"users/79357/ej-HD 1.mp4",:metadata_directive=>"COPY",:storage_class=>"STANDARD",:use_multipart_copy=>true)
[AWS S3 200 0.598405 0 retries] copy_part(:bucket_name=>"my-bucket",:copy_source=>"/my-bucket/users/79357/ej-hd-1.mp4",:copy_source_range=>"bytes=0-5242895",:key=>"users/79357/ej-HD 1.mp4",:metadata_directive=>"COPY",:part_number=>1,:storage_class=>"STANDARD",:upload_id=>"[filtered]",:use_multipart_copy=>true)
[AWS S3 200 0.406417 0 retries] copy_part(:bucket_name=>"my-bucket",:copy_source=>"/my-bucket/users/79357/ej-hd-1.mp4",:copy_source_range=>"bytes=5242896-6577314",:key=>"users/79357/ej-HD 1.mp4",:metadata_directive=>"COPY",:part_number=>2,:storage_class=>"STANDARD",:upload_id=>"[filtered]",:use_multipart_copy=>true)
[AWS S3 200 0.3649 0 retries] complete_multipart_upload(:bucket_name=>"my-bucket",:key=>"users/79357/ej-HD 1.mp4",:parts=>[{:etag=>"\"68ad4a2f5b8337f6fc450b5f27f4cc8a\"",:part_number=>1},{:etag=>"\"[filtered]\"",:part_number=>2}],:upload_id=>"[filtered]")
[AWS S3 204 0.152 0 retries] delete_object(:bucket_name=>"my-bucket",:key=>"users/79357/ej-hd-1.mp4")
=> <AWS::S3::S3Object:my-bucket/users/79357/ej-HD 1.mp4>
```

##### Version 2 (Bad on version 2.2.22)

```
> source_key = 'users/79357/ej-HD 1.mp4'
=> "users/79357/ej-HD 1.mp4"
> target_key = 'users/79357/ej-hd-1.mp4'
=> "users/79357/ej-hd-1.mp4"
> source = Aws::S3::Object.new 'my-bucket', source_key
=> #<Aws::S3::Object bucket_name="my-bucket", key="users/79357/ej-HD 1.mp4">
> target = Aws::S3::Object.new 'my-bucket', target_key
=> #<Aws::S3::Object bucket_name="my-bucket", key="users/79357/ej-hd-1.mp4">
> source.exists?
=> true
> source.move_to target, multipart_copy: true
Aws::S3::Errors::NotFound:
from /Users/e***s/.rvm/gems/ruby-2.3.0@bg/gems/aws-sdk-core-2.2.22/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'
```

Can someone please explain this behaviour? 
It seems like a regression to me.

Thanks.

**Comment by awood45** [comment]
Working on reproducing this now.

**Comment by awood45** [comment]
I ran an example, copying an object with a space to another object with a space, identical to your example, and it works in `2.3.21`. Checking if there was a bug previously patched.

**Comment by awood45** [comment]
There was a bug, which was fixed in version `2.2.32`: [commit fixing this issue](https://github.com/aws/aws-sdk-ruby/commit/0aadf59a59f20885e83e35bff8c67ea07c573aa8#diff-dae4e903b63484c6d36c2430e27ebde6).

We had a bug where we didn't properly URL encode certain characters that required URL encoding in this specific usage path. Your SDK version contains that bug, later versions do not. If you upgrade your SDK version, you should no longer see this issue.

**Comment by edjames** [comment]
DOH! Sorry, I should've tested the latest version. My apologies, and thanks very much for your prompt response.
